libflux/src/flux/semantic/mod.rs

```diff
@@ -12,8 +12,8 @@ pub mod nodes;
 pub mod types;
 pub mod walk;
 
-#[cfg(test)]
-mod parser;
+#[allow(dead_code)]
```

Comment:
I'm curious why you need to allow dead code here.  Is there something that will eventually be made public?  If that's the case, I think it's better to just add `pub` to make dead code reachable, rather than disable this warning for a whole module.

Reply:
By labeling it as `pub` does that get rid of the dead code warning. I thought everything was being used, just across modules. I'll check again.